Helper/Cart.php

```diff
@@ -315,8 +334,39 @@ public function getCartData($payment_only, $place_order_payload, $quote = null)
             return $cart;
         }
 
+        $quote->setBoltParentQuoteId($quote->getId());
+        $quote->getReservedOrderId() ?: $this->setReservedOrderId($quote);
+
+        ////////////////////////////////////////////////////////
+        // CLONE THE QUOTE and quote billing / shipping  address
+        ////////////////////////////////////////////////////////
+        $clone = $this->quoteFactory->create();
+
+        $clone->merge($quote);
+
+        foreach ($quote->getData() as $key => $value) {
```

Comment:
for magento1 we explicitly set a few extra fields. can you double check if these are not needed?

https://github.com/BoltApp/bolt-magento1/blob/094072a4037c28dac49070fc27a48cea0f7036ec/app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php#L257

Reply:
I've run the tests for anonymous and logged in user, regular and orphaned transactions, Capillus reward points too. Seems like all data is there.

The key difference between M2 and M1 implementation is that, as M2 works in a much more async fashion, not only one clone (immutable quote in M1) exists but every cart change (possibly on different tabs) triggers a new clone (bolt order) creation as a snapshot of the parent quote at the particular point in time. 

Every clone has its own shipping and billing address entities cloned (all the data copied) from the parent quote addresses. On the other hand, M1 immutable quote inherits (gets assigned) the parent address instances. I think that implies the necessity of skipping some fields and adding others explicitly to keep the totals in order. The M2 way every clone has its own totals as well as everything else.

All being said I will double check again.

Reply:
Double checked.

Reply:
@YevhenShBolt All previously saved data that may have been added by other plugins